### PR TITLE
feat(react-redux-spark): Multi widgets

### PR DESCRIPTION
**Pull request by bzang**
With some other tweaks

### CODE REVIEW

File: packages/node_modules/@ciscospark/react-redux-spark/src/spark.js [hunk 1]
```diff
@@ -10,62 +10,70 @@ import Spark from '@ciscospark/spark-core';
...
-    credentials: {
-      authorization: {
-        access_token: accessToken || credentials.access_token
-      }
-    },
-    config: {
-      logger: {
-        level: process.env.NODE_ENV === `production` ? undefined : `trace`
+  // Check if we already have a spark instance in memory
+  let ciscosparkInstance = window.ciscosparkInstance;
```
Comment (ianwremmel) [inline packages/node_modules/@ciscospark/react-redux-spark/src/spark.js]: Would it be better to attach it somewhere to `window.ciscospark`?

File: packages/node_modules/@ciscospark/react-redux-spark/src/spark.js [hunk 2]
```diff
@@ -10,62 +10,70 @@ import Spark from '@ciscospark/spark-core';
...
+
+  if (typeof ciscosparkInstance !== `object`) {
```
Comment (ianwremmel) [inline packages/node_modules/@ciscospark/react-redux-spark/src/spark.js]: `(!(ciscosparkInstance instanceof SparkCore))`?

File: packages/node_modules/@ciscospark/react-redux-spark/src/spark.js (hunk 1 above)
Comment (bzang) [inline packages/node_modules/@ciscospark/react-redux-spark/src/spark.js]: I originally wanted to but didn't want to chance the instance getting overwritten by someone attaching the SDK there.

Comment (ianwremmel) [inline packages/node_modules/@ciscospark/react-redux-spark/src/spark.js]: gotcha. I think this works then. at some point, i'd like to consolidate how a spark global instance from the sdk can coordinate, but that's probably not necessary until the widgets can run strictly on the `ciscospark` package.

**Comment by bzang** [comment]
@ianwremmel and @adamweeks , ready for round 2. Got the tests passing. Wanna get this out ASAP before Monday morning into production so we have some time to quickly tests everything in the production UI

File: packages/node_modules/@ciscospark/react-component-utils/src/metrics.js [hunk 3]
```diff
@@ -45,38 +45,39 @@ export function sendMetricsInQueue(spark) {
...
-        iteration: metricsCount[key],
-        data
-      }, fields)
-    };
-    return spark.metrics.submitClientMetrics(`${METRIC_NS}${name}`, formattedData);
-  }
-  catch (e) {
-    return console.warn(`clientMetrics: ${key} metrics was not sent`, e); // eslint-disable-line no-console
-  }
+export function sendPerfMetric({key, data, fields, tags}, spark) { // eslint-disable-line no-unused-vars
```
Comment (ianwremmel) [inline packages/node_modules/@ciscospark/react-component-utils/src/metrics.js]: Is this commented out on purpose?

**Pull request merged**